src/commands/record-traffic.ts

```diff
@@ -265,6 +270,20 @@ The app can optionally be uninstalled automatically afterwards.`;
                             title: 'Waiting for device…',
                             task: async () => ctx.analysis.platform.waitForDevice(),
                         },
+                        {
```

Comment:
If we reset the device later on, listing the apps here won’t be of much use. But I am on it.

Reply:
I did deliberately put it right at the start because `ensureDevice()` can take quite a while. And it's not nice to have interactive elements in-between long wait periods (see Windows installs :D).

But you are of course correct about the resetting. So the solution here is probably to have two separate cases.

Reply:
But `listApps` dies assume `ensureDevice` has been run. At least on Android, we’d run into issues, because the adb server might not be running, yet. I guess we could run`ensureAdb`, though…

Reply:
Well, fair enough. Nah, I guess that would be too much special-casing and prone to breaking when we change something in `ensureDevice()`.